src/pk.c

```diff
@@ -12002,9 +12002,12 @@ WOLFSSL_EC_KEY* wolfSSL_PEM_read_bio_ECPrivateKey(WOLFSSL_BIO* bio,
             err = 1;
         }
     }
-    /* Read a PEM key in to a new DER buffer. */
-    if ((!err) && (pem_read_bio_key(bio, cb, pass, ECC_PRIVATEKEY_TYPE,
-            &keyFormat, &der) <= 0)) {
+    /* Read a PEM key in to a new DER buffer.
+     * To check ENC EC PRIVATE KEY, it uses PRIVATEKEY_TYPE to call
+     * pem_read_bio_key(), and then check key format if it is EC.
+     */
+    if ((!err) && (pem_read_bio_key(bio, cb, pass, PRIVATEKEY_TYPE,
+            &keyFormat, &der) <= 0) && (keyFormat != ECDSAk)) {
```

Comment:
Put keyFormat check in new if statement.
(Easier to debug.)

Reply:
thanks. Fixed